cmd/ovirt-populator/ovirt-populator.go

```diff
@@ -0,0 +1,252 @@
+			gvr = schema.GroupVersionResource{Group: groupName, Version: apiVersion, Resource: resource}
+		)
+		populator_machinery.RunController(masterURL, kubeconfig, imageName, httpEndpoint, metricsPath,
+			namespace, prefix, gk, gvr, mountPath, devicePath, getPopulatorPodArgs)
+	case "populate":
+		populate(masterURL, kubeconfig, crName, engineUrl, secretName, diskID, fileName, namespace, crNamespace)
+	default:
+		klog.Fatalf("Invalid mode: %s", mode)
+	}
+}
+
+type engineConfig struct {
+	URL      string
+	username string
+	password string
+	ca       string
+}
+
+type TransferProgress struct {
+	Transferred uint64  `json:"transferred"`
+	Description string  `json:"description"`
+	Size        *uint64 `json:"size,omitempty"`
+	Elapsed     float64 `json:"elapsed"`
+}
+
+func getSecret(secretName, engineURL, namespace string) engineConfig {
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		klog.Fatal(err.Error())
+	}
+	// creates the clientset
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		klog.Fatal(err.Error())
+	}
+
+	secret, err := clientset.CoreV1().Secrets(namespace).Get(context.TODO(), secretName, metav1.GetOptions{})
+	if err != nil {
+		klog.Fatal(err.Error())
+	}
+
+	return engineConfig{
+		URL:      engineURL,
+		username: string(secret.Data["user"]),
+		password: string(secret.Data["password"]),
+		ca:       string(secret.Data["cacert"]),
+	}
+}
+
+func getPopulatorPodArgs(rawBlock bool, u *unstructured.Unstructured) ([]string, error) {
+	var ovirtVolumePopulator v1beta1.OvirtVolumePopulator
+	err := runtime.DefaultUnstructuredConverter.FromUnstructured(u.UnstructuredContent(), &ovirtVolumePopulator)
+	args := []string{"--mode=populate"}
+	if nil != err {
+		return nil, err
+	}
+
+	if rawBlock {
+		args = append(args, "--file-name="+devicePath)
+	} else {
+		args = append(args, "--file-name="+mountPath+"disk.img")
+	}
+
+	args = append(args, "--secret-name="+ovirtVolumePopulator.Spec.EngineSecretName)
+	args = append(args, "--disk-id="+ovirtVolumePopulator.Spec.DiskID)
+	args = append(args, "--engine-url="+ovirtVolumePopulator.Spec.EngineURL)
+	args = append(args, "--cr-name="+ovirtVolumePopulator.Name)
+	args = append(args, "--cr-namespace="+ovirtVolumePopulator.Namespace)
+
+	return args, nil
+}
+
+func populate(masterURL, kubeconfig, crName, engineURL, secretName, diskID, fileName, namespace, crNamespace string) {
+	engineConfig := getSecret(secretName, engineURL, namespace)
+
+	// Write credentials to files
+	ovirtPass, err := os.Create("/tmp/ovirt.pass")
+	if err != nil {
+		klog.Fatalf("Failed to create ovirt.pass %s", err)
+	}
+
+	defer ovirtPass.Close()
+	if err != nil {
+		klog.Fatalf("Failed to create file %s", err)
+	}
+	ovirtPass.Write([]byte(engineConfig.password))
+
+	cert, err := os.Create("/tmp/ca.pem")
+	if err != nil {
+		klog.Fatalf("Failed to create ca.pem %s", err)
+	}
+
+	defer cert.Close()
+	if err != nil {
+		klog.Fatalf("Failed to create file %s", err)
+	}
```

Comment:
and what did you mean to check here?

Reply:
still there's no assignment to `err` after it was checked above

Reply:
correct, i forgot the assignment.